user/serializers.py

```diff
@@ -2,0 +1,29 @@
+from rest_framework import serializers
+from django.contrib.auth import get_user_model
+
+
+class UserSerializer(serializers.ModelSerializer):
+    def validate(self, attrs):
+        data = super(UserSerializer, self).validate(attrs)
+        get_user_model().validate_password(
+            attrs["password"],
+            serializers.ValidationError
+        )
+        return data
+
+    class Meta:
+        model = get_user_model()
+        fields = ("id", "username", "email", "password", "is_staff")
+        read_only_fields = ("id", "is_staff")
+        extra_kwargs = {"password": {"write_only": True, "min_length": 5}}
+
+    def create(self, validated_data):
+        return get_user_model().objects.create_user(**validated_data)
+
+    def update(self, instance, validated_data):
+        password = validated_data.pop("password", None)
```

Comment:
In the update method, you're popping the password from validated_data. This means that if the password isn't in validated_data, it will be removed and no error will be raised. However, if the password is a required field, you should use validated_data['password'] instead of validated_data.pop('password', None). If the password isn't in validated_data, this will raise a KeyError.